### PR TITLE
Agent docs and config improvements

### LINKED ISSUE #2113: Agent docs/config improvements after EthTokyo
Problems:
            - A lot of configuration fatigue
            - missing field `validator` is too vague
            - No such file or directory (os error 2) is a frustrating error
            - Machine specs section on starting a validator looks out of place
            - `Error: Destination chain base does not have a configured signer` feels not as actionable as we can be
            - `Missing chain configuration for sepolia; this includes protocol type and the connection information` is quite vague
            - in the relayer guide, similar to the validator guide, the specific sequence is a bit unclear. Going page by page gets me to message filtering which i should not need to check out. The Env page does again not distinguish between required and optional vars and does not tell me what the next step is, when i am done
            - Way too easy to miss `HYP_RELAYER_ALLOWLOCALCHECKPOINTSYNCERS`
            - Not mounting the DB_PATH will result in relayers needing to index from genesis every time which is very annoying
            - When running two relayers locally, it is possible to actually have both read/write to the same DB which is very hard to debug and obviously does not work
            -  "no name found" - typo results in weird message
            - When restarting anvil, previous artifacts are no longer valid, but very hard to debug

Possible Solutions (not exhaustive):
            - Collapse running validator/relayer into a single page as much as possible
            - Default validator interval to something like 5s
            - Default HYP_BASE_METRICS (maybe already done)
            - Default HYP_BASE_TRACING_LEVEL and FORMAT (maybe already done)
            - Segment the environment variables/config into required and optional ones (i.e. ones that have reasonable defaults should be further down)
            - “missing field `validator`” should have a more useful message indicate to specify the validator key
            - When using localStorage checkpoint synced, just create the directory if it doesn’t exist
            - When signers are missing, point to default signer as much as possible
            - If an RPC url is missing, call it out as such, don’t say chain configuration is missing with protocol type and connection information
            - HYP_RELAYER_ALLOWLOCALCHECKPOINTSYNCERS needs to be made more obvious. When it’s not enabled but we discover a local storage checkpoint synced, we should log an info log line to indicate that this is the issue why metadata could not be fetched
            - Indicate that relayers when using docker should mount external storage for DB_PATH
            - Detect when the agent runs on a DB for which it was not configured (i.e. its the DB of some other agents)
            - "no name found" - should have a more clear error message that they probably have a typo in the RPC url env var

### PR DESCRIPTION
**Pull request by mattiecnvr**
### Description

Bunch of minor tweaks and updates based on what we learned from EthTokyo.

- Updated Rust readme and the root readme to align with the hyperlane docs
- Defaults to a 5 sec validator interval now
- Makes the metrics port default to `9090` and have a soft failure if it can't start
- Now will automatically create the checkpointsyncer directory if it is missing
- Hyperlane database now officially uses the domain name and id as a prefix internally to support multiple relayers using the same database
- 

### Drive-by changes

Refactored and cleaned up the database code. Removed a lot of duplicate/unnecessary/redundant functions.

### Related issues

- Fixes #2113 

### Backward compatibility

_Are these changes backward compatible?_

Yes

_Are there any infrastructure implications, e.g. changes that would prohibit deploying older commits using this infra tooling?_

Yes - This changes the database prefixes internally to use domain name and domain id; also prepares us for #1899 

### Testing

_What kind of testing have these changes undergone?_

Manual
Unit Tests

### CODE REVIEW

File: rust/README.md [hunk 1]
```diff
@@ -5,7 +5,8 @@
 - install `rustup`
   - [link here](https://rustup.rs/)
 
-Note: You should be running >= version `1.68.0` of the rustc compiler, you can see that version with this command and should see similar output:
+Note: You should be running >= version `1.68.0` of the rustc compiler, you can see that version with this command and
```
Comment (asaj) [inline rust/README.md:8]: Update the output to match the version requirement

File: rust/README.md [hunk 2]
```diff
@@ -14,6 +15,71 @@ info: This is the version for the rustup toolchain manager, not the rustc compil
...
+production environment.
+
+```bash
+cd rust
+./build.sh <image_tag>
+```
+
+### Deploy Procedure
+
+The contract addresses of each deploy can be found in `rust/config`. The latest
```
[the remark this answers is not in the document]
Comment (asaj) [reply rust/README.md:69]: Actually this whole paragraph looks wrong

File: rust/hyperlane-base/src/db/hyperlane_db.rs [hunk 3]
```diff
@@ -5,15 +5,18 @@ use tokio::time::sleep;
...
+    HyperlaneDomain, HyperlaneMessage, InterchainGasExpenditure, InterchainGasPayment,
+    InterchainGasPaymentMeta, LogMeta, H256, U256,
 };
 
 use super::{
     storage_types::{InterchainGasExpenditureData, InterchainGasPaymentData},
     DbError, TypedDB, DB,
 };
 
+// these keys MUST not be given multiple uses in case multiple agents are
```
Comment (asaj) [inline rust/hyperlane-base/src/db/hyperlane_db.rs:17]: I'm not sure I follow this comment

File: rust/hyperlane-base/src/db/mod.rs [hunk 4]
```diff
@@ -95,83 +95,17 @@ impl DB {
...
         Ok(self.0.put(key, value)?)
     }
 
     /// Retrieve a value from the DB
-    fn _retrieve(&self, key: impl AsRef<[u8]>) -> Result<Option<Vec<u8>>> {
+    pub fn retrieve(&self, key: &[u8]) -> Result<Option<Vec<u8>>> {
         Ok(self.0.get(key)?)
     }
 
-    /// Prefix a key and store in the DB
```
Comment (asaj) [inline rust/hyperlane-base/src/db/mod.rs:107]: Unused, I presume?

File: rust/hyperlane-base/src/db/mod.rs [hunk 5]
```diff
@@ -95,83 +95,17 @@ impl DB {
...
-        &self,
-        prefix: impl AsRef<[u8]>,
-        key: &K,
-    ) -> Result<Option<V>> {
-        self.retrieve_decodable(prefix, key.to_vec())
-    }
-
     /// Get prefix db iterator for `prefix`
-    pub fn prefix_iterator(&self, prefix: impl AsRef<[u8]>) -> DBIterator {
+    pub fn prefix_iterator(&self, prefix: &[u8]) -> DBIterator {
```
Comment (asaj) [inline rust/hyperlane-base/src/db/mod.rs:108]: What is this used for?

File: rust/hyperlane-base/src/db/typed_db.rs [hunk 6]
```diff
@@ -1,17 +1,15 @@
...
-use hyperlane_core::{Decode, Encode};
+use hyperlane_core::{Decode, Encode, HyperlaneDomain};
 
 use crate::db::{DbError, DB};
 
 type Result<T> = std::result::Result<T, DbError>;
 
 /// DB handle for storing data tied to a specific type/entity.
 ///
-/// Key structure: ```<type_prefix>_<additional_prefix(es)>_<key>```
```
Comment (asaj) [inline rust/hyperlane-base/src/db/typed_db.rs:11]: What was the previous "type_prefix", and why is it okay to remove?

File: rust/hyperlane-base/src/db/typed_db.rs [hunk 7]
```diff
@@ -22,12 +20,27 @@ impl AsRef<DB> for TypedDB {
...
+            .iter()
+            .chain(b"_")
+            .chain(&domain.id().to_be_bytes())
+            .chain(b"_")
+            .copied()
+            .collect();
+        Self { domain_prefix, db }
+    }
+
+    fn full_key(&self, prefix: &[u8], key: &[u8]) -> Vec<u8> {
```
Comment (asaj) [inline rust/hyperlane-base/src/db/typed_db.rs:37]: nit: prefixed_key?

File: rust/hyperlane-base/src/db/typed_db.rs [hunk 8]
```diff
@@ -57,8 +76,7 @@ impl TypedDB {
         key: &K,
         value: &V,
     ) -> Result<()> {
-        self.db
-            .store_keyed_encodable(self.full_prefix(prefix), key, value)
+        self.store_encodable(prefix, key.to_vec(), value)
```
Comment (asaj) [inline rust/hyperlane-base/src/db/typed_db.rs:79]: Should this function be public? Feels like a potential footgun to have public functions that don't handle the key prefixing

Comment (asaj) [reply rust/hyperlane-base/src/db/typed_db.rs:79]: Especially since full_key is private rn

File: rust/hyperlane-base/src/settings/base.rs [hunk 9]
```diff
@@ -49,7 +49,7 @@ pub struct Settings {
     /// Configuration for contracts on each chain
     pub chains: HashMap<String, ChainConf>,
     /// Port to listen for prometheus scrape requests
-    pub metrics: Option<u16>,
+    pub metrics: u16,
```
Comment (asaj) [inline rust/hyperlane-base/src/settings/base.rs:52]: nit: If it's easy, suggest metrics_port

File: rust/hyperlane-base/src/db/typed_db.rs (hunk 6 above)
Comment (mattiecnvr) [reply rust/hyperlane-base/src/db/typed_db.rs:11]: It was poorly named, it was only being used as the domain name. This is not a change to logic, just clarity.

File: rust/hyperlane-base/src/settings/base.rs (hunk 9 above)
Comment (mattiecnvr) [reply rust/hyperlane-base/src/settings/base.rs:52]: This one is easy, changing the name in the actual config is not.

File: rust/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply rust/README.md]: Fixed.

File: rust/hyperlane-base/src/db/typed_db.rs (hunk 7 above)
Comment (mattiecnvr) [reply rust/hyperlane-base/src/db/typed_db.rs]: Changed.

File: rust/hyperlane-base/src/db/typed_db.rs (hunk 8 above)
Comment (mattiecnvr) [reply rust/hyperlane-base/src/db/typed_db.rs:79]: This function requires the prefix as an argument. I don't understand your comment.

File: rust/hyperlane-base/src/db/hyperlane_db.rs (hunk 3 above)
Comment (asaj) [reply rust/hyperlane-base/src/db/hyperlane_db.rs:17]: I still don't follow, can we follow up on discord?

File: rust/hyperlane-base/src/db/mod.rs (hunk 5 above)
Comment (asaj) [reply rust/hyperlane-base/src/db/mod.rs:108]: And what do we use that for?

File: rust/hyperlane-base/src/db/typed_db.rs (hunk 8 above)
Comment (asaj) [reply rust/hyperlane-base/src/db/typed_db.rs:79]: Tbh coming back to this I don't think I do either. My recollection was that some public functions took a prefixed key, and others did not, but it looks like every public fn takes a prefix and a key

**Review by asaj — APPROVED** [review]

**Pull request merged**